File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> sum = new Dictionary<int, int>(); 

        for(int i =0;i< nums.Length; i++){
            int currentTotal =   target - nums[i]; 
            if(sum.ContainsKey(currentTotal))
            {
                return new int[] { sum[currentTotal], i};
            }
            sum[nums[i]]=i;
        }

        return new int[] {};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sum ={}

        for idx in range(len(nums)):
            current = target - nums[idx]
            if current in sum:
                return [sum[current], idx]
            sum[nums[idx]] = idx

        return []

        